marimo/_runtime/runner/hooks_preparation.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Marimo. All rights reserved.
+from marimo._runtime import dataflow
+from marimo._runtime.runner import cell_runner
+
+
+def _update_stale_statuses(runner: cell_runner.Runner) -> None:
+    if runner.execution_mode == "detect":
+        for cid in dataflow.transitive_closure(
+            runner.graph, set(runner.cells_to_run), inclusive=False
+        ):
+            runner.graph.cells[cid].set_stale(stale=True)
+
+    for cid in runner.cells_to_run:
+        if runner.graph.is_disabled(cid):
+            runner.graph.cells[cid].set_stale(stale=True)
```

Comment:
in "detect", could this `.set_stale(stale=True)` twice? probably a pre-optimization, but could force re-renders on the frontend

Reply:
it could ... but I _think_ this code path won't be hit when the mode is `detect`.